src/test/shell/bazel/bazelignore_test.sh

```diff
@@ -206,4 +206,16 @@ EOI
     assert_not_contains "ignoreme" output
 }
 
+test_invalid_path() {
+    rm -rf work && mkdir work && cd work
+    create_workspace_with_default_repos WORKSPACE
+    echo -e "foo/\0/bar" > .bazelignore
+    echo 'filegroup(name="f", srcs=glob(["**"]))' > BUILD
+    if bazel build //... 2> "$TEST_log"; then
+      fail "Bazel build should have failed"
+    fi
+    expect_log "java.nio.file.InvalidPathException: Nul character not allowed"
+    bazel shutdown
```

Comment:
I'm not super happy about this.

If you remove it, a different test in this file fails which doesn't fail individually:

```
** test_symlink_cycle_ignored **************************************************
WORKSPACE
-- Test log: -----------------------------------------------------------
$TEST_TMPDIR defined: output root default is '/private/var/tmp/_bazel/57872f73c892b24b35be334f00d21dae/sandbox/darwin-sandbox/4132/execroot/_main/_tmp/205f62347f64a58315b97e6ebfa13f59' and max_idle_secs default is '15'.
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
/etc/bazel.bazelrc
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.NullPointerException: Cannot invoke "com.google.devtools.build.lib.skyframe.IgnoredPackagePrefixesValue.getPatterns()" because "ignoredPackagePrefixesValue" is null
	at com.google.devtools.build.lib.skyframe.SkyframeExecutor.handleDiffs(SkyframeExecutor.java:3181)
	at com.google.devtools.build.lib.skyframe.SequencedSkyframeExecutor.sync(SequencedSkyframeExecutor.java:282)
	at com.google.devtools.build.lib.runtime.CommandEnvironment.syncPackageLoading(CommandEnvironment.java:752)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:582)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:250)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:604)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$1(GrpcServerImpl.java:676)
	at io.grpc.Context$1.run(Context.java:566)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Adding this fixes it which means that my change makes it so that Bazel doesn't crash when it encounters this issue. But, it might crash on the next run unless it's restarted after the `.bazelignore` is fixed.

Open to thoughts on what might be going wrong here.